lib/src/repository/traits/lock_repository.dart

```diff
@@ -21,4 +42,33 @@ class LockRepository {
       throw errors.first;
     }
   }
+
+  static LockTrait getLockTrait(dynamic trait) {
+    if (trait is GgetDeviceData_device_traits__asLockDeviceTrait ||
+        trait
+            is GgetDevicesData_me_devices_edges_node_traits__asLockDeviceTrait) {
+      final properties = [
+        SupportsIsJammed(trait.properties.supportsIsJammed ?? false)
```

Comment:
I haven't 100% decided if it makes a particular difference but realistically these properties should only exist once for a trait.